### PR TITLE
Fix bio editing bug

### PR DESCRIPTION
**Pull request by a-bainbridge**

### CODE REVIEW

File: server/routes/index.ts [hunk 1]
```diff
@@ -70,8 +70,8 @@ function createRouter(repositories: Repositories) {
             res.status(403).send(`You do not have permission to edit user id ${req.targetUser!.id}`);
             return next();
         }
-        req.user.updateFromData(req.body);
-        users.save(req.user)
+        ((req as any).targetUser as User).updateFromData(req.body);
+        users.save(((req as any).targetUser as User))
```
Comment (amikhalev) [inline server/routes/index.ts]: I added targetUser to the type of req, so you shouldn't have to cast it

**Review by amikhalev — CHANGES_REQUESTED** [review]

File: server/routes/index.ts [hunk 2]
```diff
@@ -70,12 +70,14 @@ function createRouter(repositories: Repositories) {
             res.status(403).send(`You do not have permission to edit user id ${req.targetUser!.id}`);
             return next();
         }
-        req.user.updateFromData(req.body);
-        users.save(req.user)
-            .then(() => {
-                res.redirect(req.targetUser!.url);
-                next();
+        req.targetUser!.updateFromData(req.body);
+        if (req.targetUser) {
```
Comment (amikhalev) [inline server/routes/index.ts]: You are calling a method on targetUser above. How could it possible be null here?

`!.` is not an operator. `req.targetUser!` asserts that `req.targetUser` is not null

**Review by amikhalev — CHANGES_REQUESTED** [review]

File: server/routes/index.ts (hunk 2 above)
Comment (a-bainbridge) [reply server/routes/index.ts]: error TS2345: Argument of type 'User | undefined' is not assignable to parameter of type 'DeepPartial<User>'.

Comment (amikhalev) [reply server/routes/index.ts]: req.targetUser! asserts that req.targetUser is not null

**Review by amikhalev — CHANGES_REQUESTED** [review]

File: server/routes/index.ts (hunk 2 above)
Comment (a-bainbridge) [reply server/routes/index.ts]: go ahead and compile it without the if statement then

Comment (amikhalev) [reply server/routes/index.ts]: image
Works fine for me. When I assert that `req.targetUser` is not null

**Review by amikhalev — CHANGES_REQUESTED** [review]

File: server/routes/index.ts (hunk 2 above)
Comment (amikhalev) [reply server/routes/index.ts]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Comment (amikhalev) [reply server/routes/index.ts]: Sorry if I was unclear, but it does the assertion at the expression level. So you have to do it every time you use a variable that has a type `undefined | T` or `null | T`

Comment (a-bainbridge) [reply server/routes/index.ts]: Oh, i'm just now seeing the assertion you added to save. Apologies for my confusion

**Review by amikhalev — APPROVED** [review]

**Pull request merged**